systems/plants/RigidBodyManipulator.cpp

```diff
@@ -3239,14 +3300,82 @@ int RigidBodyManipulator::findLinkId(string linkname, int robot)
   if(num_match != 1)
   {
     cerr<<"couldn't find unique link "<<linkname<<endl;
-    return(EXIT_FAILURE);
+    return(nullptr);
   }
   else
   {
-    return ind_match;
+    return this->bodies[ind_match];
   }
 }
 
+int RigidBodyManipulator::findLinkId(string name, int robot)
+{
+	shared_ptr<RigidBody> link = findLink(name,robot);
+	if (link == nullptr) return(EXIT_FAILURE);
```

Comment:
good call.  that is a hold-over from the existing code, but would have led to very mysterious behavior.  (on my mac EXIT_FAILURE = 1, which is a valid link id)